src/lava/magma/runtime/runtime_service.py

```diff
@@ -173,73 +176,78 @@ def run(self):
                     return
                 elif np.array_equal(command, MGMT_COMMAND.PAUSE):
```

Comment:
We have to go into HOST phase when PAUSE is received

Reply:
We should be always in the HOST phase if TERMINATED or PAUSE can be received. The while loop in else only breaks if we reach the HOST phase and we start in the HOST phase.
Or am I missing something?

Reply:
No, you are right. 

The issue is actually not here but in the else branch. As part of a proper pause() implementation, we should check occasionally, if a PAUSE has been received. At this point we should break out at the next possible end of time step.

Reply:
Right, this should then be part of the PAUSE implementation (Issue #53)